.github/workflows/reusable_image_release.yaml

```diff
@@ -0,0 +1,251 @@
+  ORIGINAL_REPO_NAME: ${{ inputs.original_repo_name }}
+  REPO_FULL_NAME: ${{ github.event.repository.full_name }}
+  INTEGRATION: ${{ inputs.integration_name }}
+  S3_PATH: ${{ inputs.s_three_path }}
+  S3_BASE_FOLDER: ${{ inputs.s_three_folder }}
+
+jobs:
+  test-nix:
+    if: ${{ inputs.run_nix_unit_tests }}
+    uses: newrelic/coreint-automation/.github/workflows/_test_nix.yaml@v3
+
+  test-windows:
+    if: ${{ inputs.run_windows_unit_tests }}
+    runs-on: windows-latest
+    env:
+      GOPATH: ${{ github.workspace }}
+    defaults:
+      run:
+        working-directory: src/github.com/${{env.ORIGINAL_REPO_NAME}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: src/github.com/${{env.ORIGINAL_REPO_NAME}}
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: "src/github.com/${{ env.ORIGINAL_REPO_NAME }}/go.mod"
+      - name: Running unit tests
+        shell: pwsh
+        run: |
+          .\build\windows\unit_tests.ps1
+    
+  integration-tests:
+    name: 🚧 Integration Tests
+    if: ${{ inputs.run_integration_tests }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [arm64, amd64]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.docker_username }}
+          password: ${{ secrets.docker_password }}
+      - name: Running integration tests
+        run: |
+          make integration-tests-${{ matrix.arch }}
+
+  release-integration:
+    name: Publish container images to registry
+    needs: [test-nix, test-windows, integration-tests]
+    if: |
+      always() &&
+      (needs.integration-tests.result == 'success' || needs.integration-tests.result == 'skipped') &&
+      (needs.test-nix.result == 'success' || needs.test-nix.result == 'skipped') &&
+      (needs.test-windows.result == 'success' || needs.test-windows.result == 'skipped')
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: ${{ inputs.go_version_file }}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.docker_username }}
+          password: ${{ secrets.docker_password }}
+
+      - name: Add pre-release tag suffix
+        if: ${{ github.event.release.prerelease }}
+        run: |
+            echo "TAG_SUFFIX=-pre" >> $GITHUB_ENV
```

Comment:
some repos are using a different envVar name, therefore this is totally ignore in some cases